### PR TITLE
SALTO-3986: Support Brand Email template

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Added support for email customizations

---

_Additional context for reviewer_

- Each brand has a fixed type of `EmailTemplate` instances, for each type of email ("forget password", "user activation", etc). The email templates are auto created when the brand is added (similar to brand theme)
- For each type of email, Okta allows customizing it in different languages.
- Planned to add E2E, but our E2E account doesn't allow customizations. Will update tests in `adapter.test.ts` soon.

---
_Release Notes_: 

_Okta adapter_:
- Add support in Email Templates and customizations for Brands

---
_User Notifications_: 

_Okta adapter_:
- `EmailCustomization` and `EmailTemplate` elements will be added for workspaces with brands customizations.

### CODE REVIEW

File: packages/okta-adapter/src/definitions/deploy/deploy.ts [hunk 1]
```diff
@@ -840,6 +844,106 @@ const createCustomizations = (): Record<string, InstanceDeployApiDefinitions> =>
...
+                transformation: {
+                  root: 'settings',
+                },
+              },
+              condition: { skipIfIdentical: true, transformForCheck: { root: 'settings' } },
+            },
+          ],
+          // EmailTemplates are removed automatically by Okta when the Brand is removed.
+          // A separate Change Validator ensures that templates aren't removed by themselves.
+          remove: [{ request: { earlySuccess: true } }],
```
Comment (Nurdok) [inline packages/okta-adapter/src/definitions/deploy/deploy.ts:893]: In other types that are automatically removed (`ProfileMapping`, `BrandTheme`) we perform a GET to make sure that the instance was deleted. Should we do this here, too?

File: packages/okta-adapter/src/definitions/deploy/deploy.ts [hunk 2]
```diff
@@ -840,6 +844,106 @@ const createCustomizations = (): Record<string, InstanceDeployApiDefinitions> =>
...
+        },
+      },
+      toActionNames: ({ change }) => (isAdditionChange(change) ? ['add', 'modify'] : [change.action]),
+      actionDependencies: [{ first: 'add', second: 'modify' }],
+    },
+    EmailCustomization: {
```
Comment (Nurdok) [inline packages/okta-adapter/src/definitions/deploy/deploy.ts:899]: Please put this in a `const` too.

File: packages/okta-adapter/src/definitions/fetch/fetch.ts [hunk 3]
```diff
@@ -705,6 +718,83 @@ const createCustomizations = ({
...
   },
+  [EMAIL_TEMPLATE]: {
+    requests: [
+      {
+        endpoint: {
+          path: '/api/v1/brands/{brandId}/templates/email',
+          queryArgs: { expand: 'settings' },
+        },
+        transformation: {
+          rename: [{ from: '_embedded.settings', to: 'settings', onConflict: 'override' }],
```
Comment (Nurdok) [inline packages/okta-adapter/src/definitions/fetch/fetch.ts:729]: nice

File: packages/okta-adapter/e2e_test/adapter.test.ts
Comment (Nurdok) [inline packages/okta-adapter/e2e_test/adapter.test.ts:1]: Please also add some HTTP-recording tests (the other `adapter.test.ts` 😄)

**Review by Nurdok — CHANGES_REQUESTED** [review]
Looks great! That was really fast!

File: packages/okta-adapter/src/definitions/deploy/deploy.ts (hunk 1 above)
Comment (shir-reifenberg) [reply packages/okta-adapter/src/definitions/deploy/deploy.ts:893]: true, wanted to implement it with `additionalStatusCode` but I think there's might be an issue with it (opened SALTO-6664), I think we can start like this and adjust once fixed

File: packages/okta-adapter/e2e_test/adapter.test.ts
Comment (shir-reifenberg) [reply packages/okta-adapter/e2e_test/adapter.test.ts:1]: Added 😄 seems like i'm doing something wrong cause the entire thing fails in my added tests, let's discuss tomorrow IRL

**Review by Nurdok — APPROVED** [review]

**Pull request merged**